modules/adbutlerBidAdapter.js

```diff
@@ -1,130 +1,122 @@
+      if (minCPM !== '' && CPM < parseFloat(minCPM)) {
+        isCorrectCPM = false;
+      }
+      if (maxCPM !== '' && CPM > parseFloat(maxCPM)) {
+        isCorrectCPM = false;
+      }
 
-    // Define callback function for bid responses
-    $$PREBID_GLOBAL$$.adbutlerCB = function(aBResponseObject) {
-      var bidResponse = {};
-      var callbackID = aBResponseObject.zone_id + '_' + aBResponseObject.place;
-      var width = parseInt(aBResponseObject.width);
-      var height = parseInt(aBResponseObject.height);
-      var isCorrectSize = false;
-      var isCorrectCPM = true;
-      var CPM;
-      var minCPM;
-      var maxCPM;
-      var bidObj = callbackData[callbackID] ? utils.getBidRequest(callbackData[callbackID].bidId) : null;
-
-      if (bidObj) {
-        if (aBResponseObject.status === 'SUCCESS') {
-          CPM = aBResponseObject.cpm;
-          minCPM = utils.getBidIdParameter('minCPM', bidObj.params);
-          maxCPM = utils.getBidIdParameter('maxCPM', bidObj.params);
-
-          // Ensure response CPM is within the given bounds
-          if (minCPM !== '' && CPM < parseFloat(minCPM)) {
-            isCorrectCPM = false;
-          }
-          if (maxCPM !== '' && CPM > parseFloat(maxCPM)) {
-            isCorrectCPM = false;
-          }
-
-          // Ensure that response ad matches one of the placement sizes.
-          utils._each(bidObj.sizes, function(size) {
-            if (width === size[0] && height === size[1]) {
-              isCorrectSize = true;
-            }
-          });
-
-          if (isCorrectCPM && isCorrectSize) {
-            bidResponse = bidfactory.createBid(1, bidObj);
-            bidResponse.bidderCode = 'adbutler';
-            bidResponse.cpm = CPM;
-            bidResponse.width = width;
-            bidResponse.height = height;
-            bidResponse.ad = aBResponseObject.ad_code;
-            bidResponse.ad += addTrackingPixels(aBResponseObject.tracking_pixels);
-          } else {
-            bidResponse = bidfactory.createBid(2, bidObj);
-            bidResponse.bidderCode = 'adbutler';
-          }
-        } else {
-          bidResponse = bidfactory.createBid(2, bidObj);
-          bidResponse.bidderCode = 'adbutler';
+      // Ensure that response ad matches one of the placement sizes.
+      utils._each(bidObj.sizes, function (size) {
+        if (width === size[0] && height === size[1]) {
+          isCorrectSize = true;
         }
-
-        bidmanager.addBidResponse(bidObj.placementCode, bidResponse);
+      });
+      if (isCorrectCPM && isCorrectSize) {
+        bidResponse.requestId = bidObj.bidId;
+        bidResponse.bidderCode = spec.code;
+        bidResponse.creativeId = serverResponse.placement_id;
+        bidResponse.cpm = CPM;
+        bidResponse.width = width;
+        bidResponse.height = height;
+        bidResponse.ad = serverResponse.ad_code;
+        bidResponse.ad += spec.addTrackingPixels(serverResponse.tracking_pixels);
```

Comment:
Adapters shouldn't add tracking pixels during the auction anymore. Implement [getUserSyncs](https://github.com/prebid/Prebid.js/blob/master/src/adapters/bidderFactory.js#L50) instead. `prebid-core` will make sure those pixels get added to the page alongside everyone else's

Reply:
Thanks for the info @dbemiller. We offer our customers the option to track impressions on the client-side. If they've chosen this, their impression pixel would have been added here. Just to verify, if the publisher has userSync disabled, this means there is a chance their impression wouldn't be counted in this way. Is that correct?

Reply:
Correct. Since tracking pixels affect the publisher's page performance, the prebid leadership decided that they should have the ultimate decision in whether or not to allow them.

That said, most publishers leave the default options... so it's not likely to impact you too much.

Reply:
I understand the desire to improve publisher page performance. My only concern is that when all syncs are blocked, the winning bidder will not have any way to know that they have won a particular auction. Is there any way for us to provide Prebid with a URL that can be pinged or placed if our bid has won the auction?

Reply:
I think there might be some confusion here. It's acceptable to add a tracking pixel to a bid response. This will only fire if the bid wins the auction. I think that's what this is so should be fine.

Reply:
Ah, thanks for the clarification. Yeah, this function just appends the tracking pixel(s) to the ad code in the bid response.